crates/transcode/src/decode.rs

```diff
@@ -177,7 +177,7 @@ impl<'a> Decoder<'a> {
         variant_type: &TypeDefVariant<PortableForm>,
         input: &mut &[u8],
     ) -> Result<Value> {
-        let discriminant = input.read_byte()?;
+        let discriminant = input.read_byte().unwrap_or(0);
```

Comment:
If it is an enum then the discriminant should always be the first byte.

Reply:
Is it possible to have a 0-byte enum, where the discriminant is missing? I faced this issue while testing the storage.

Reply:
Ok, yes it is possible if the enum has no variants at all: https://github.com/paritytech/parity-scale-codec/blob/master/derive/src/encode.rs#L314. 

So looks like we would just return an error from here in that case? https://github.com/paritytech/cargo-contract/blob/890d4272accd1de96828cb88e3dc04042ea8b396/crates/transcode/src/decode.rs#L186

Reply:
I am not sure. If it is allowed, why would we want to return an error while decoding storage?

Reply:
You are correct; this was a side issue of decoding `Cell<Option>`, which is not present in the storage but is defined in the type definition.